### DISCUSSION TITLE
问一下如何整合vue、vue-router、想搞个界面出来玩玩

### DISCUSSION THREAD

**Discussion by SaoXuan** [takayama-lily/oicq#357] [Q&A]

**Comment by SaoXuan** [comment]
想整合electron搞个界面出来，然后引入在vue项目中引入oicq后就会出现下面的问题（node比较菜，感觉用两个node项目无限套娃了）
![image](https://user-images.githubusercontent.com/49980280/150687085-a728935a-efe4-4298-b4c0-1e40611163b9.png)

**Comment by TankNee** [comment]
要写在主进程里吧？或者用preload暴露到渲染进程。

**Comment by koushiyu** [comment]
允许electron 使用require，我在index.html里面写了 window.oicq = require("oicq")

**Comment by so1ve** [comment]
首先，前端估计是没法直接用oicq，你要在后端写好借口通过ipc暴露给前端
这个报错就是你的webpack loader版本太低，没法解析没带括号的catch语句和static块

**Comment by MiraHikari** [comment]
前端没有能力直接调 oicq, 建议在主进程里面监听之后推到 IPC 里面.
